aea/cli/registry/fetch.py

```diff
@@ -17,38 +17,32 @@
 #
 # ------------------------------------------------------------------------------
 """Methods for CLI fetch functionality."""
-from typing import Union
 
 import click
 import os
-
-from distutils.dir_util import copy_tree
 from shutil import rmtree
 
-from aea.cli.common import DEFAULT_REGISTRY_PATH
+from aea.cli.common import Context
 from aea.cli.registry.utils import (
     request_api, download_file, extract, fetch_package
 )
 from aea.configurations.base import PublicId
 
 
-def fetch_agent(public_id: Union[PublicId, str]) -> None:
+def fetch_agent(ctx: Context, public_id: PublicId) -> None:
```

Comment:
We don't need the whole context object here. It's enough to pass `ctx.cwd` as `cwd` argument.
And by the way what a difference between `os.getcwd()` and `ctx.cwd`?

Reply:
I prefer passing the whole object as it makes it easier to read the code and extend later in case you need more from context or need to write to context